app/models/orchestration.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "velum/salt"
+require "velum/salt_orchestration"
+
+# Orchestration represents a salt orchestration event
+class Orchestration < ApplicationRecord
+  class OrchestrationAlreadyRan < StandardError; end
+
+  enum kind: [:bootstrap, :upgrade]
+  enum status: [:in_progress, :succeeded, :failed]
+
+  after_create :update_minions
+
+  # rubocop:disable Rails/SkipsModelValidations
+  def run
+    raise OrchestrationAlreadyRan unless jid.blank?
+    update_column :status, Orchestration.statuses[:in_progress]
+    _, job = case kind
+             when "bootstrap"
+               Velum::Salt.orchestrate
+             when "upgrade"
+               Velum::Salt.update_orchestration
+    end
+    update_column :jid, job["return"].first["jid"]
+    true
+  end
+  # rubocop:enable Rails/SkipsModelValidations
+
+  def self.run(kind: :bootstrap)
+    Orchestration.create(kind: kind).tap(&:run)
```

Comment:
This assumes that create will always work, which might not be true in the future if we add validations to it. Maybe you can replace it with `create!`, so it fails more clearly ?